### ISSUE TITLE
AttributeError: 'list' object has no attribute 'get_virtual_bits'

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit-terra#7662]
### Environment

- **Qiskit Terra version**: 1ae663d80295dce1b6ff25cbf29c640bf2ff9952
- **Python version**: 3.9.6
- **Operating system**: RHEL 8.5

### What is happening?

Passing a pass manager to `transpile` gives an error.

### How can we reproduce the issue?

```python
from qiskit import QuantumCircuit, transpile
from qiskit.test.mock import FakeMumbai
from qiskit.transpiler import PassManagerConfig
from qiskit.transpiler.preset_passmanagers import level_3_pass_manager

backend = FakeMumbai()
qubits = [0]

circuit = QuantumCircuit(1)
circuit.x(0)

pass_manager_config = PassManagerConfig(initial_layout=qubits, backend_properties=backend.properties())
pass_manager = level_3_pass_manager(pass_manager_config)

transpiled = transpile(circuit, pass_manager=pass_manager, unitary_synthesis_method=None)
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [1], in <module>
     12 pass_manager_config = PassManagerConfig(initial_layout=qubits, backend_properties=backend.properties())
     13 pass_manager = level_3_pass_manager(pass_manager_config)
---> 15 transpiled = transpile(circuit, pass_manager=pass_manager, unitary_synthesis_method=None)

File ~/projects/qiskit-terra/qiskit/compiler/transpiler.py:287, in transpile(circuits, backend, basis_gates, inst_map, coupling_map, backend_properties, initial_layout, layout_method, routing_method, translation_method, scheduling_method, instruction_durations, dt, approximation_degree, timing_constraints, seed_transpiler, optimization_level, pass_manager, callback, output_name, unitary_synthesis_method, unitary_synthesis_plugin_config, target)
    264     _check_conflicting_argument(
    265         optimization_level=optimization_level,
    266         basis_gates=basis_gates,
   (...)
    277         target=target,
    278     )
    280     warnings.warn(
    281         "The parameter pass_manager in transpile is being deprecated. "
    282         "The preferred way to tranpile a circuit using a custom pass manager is"
   (...)
    285         stacklevel=2,
    286     )
--> 287     return pass_manager.run(circuits, output_name=output_name, callback=callback)
    289 if optimization_level is None:
    290     # Take optimization level from the configuration or 1 as default.
    291     config = user_config.get_config()

File ~/projects/qiskit-terra/qiskit/transpiler/passmanager.py:225, in PassManager.run(self, circuits, output_name, callback)
    223     return self._run_single_circuit(circuits, output_name, callback)
    224 if len(circuits) == 1:
--> 225     return self._run_single_circuit(circuits[0], output_name, callback)
    226 return self._run_several_circuits(circuits, output_name, callback)

File ~/projects/qiskit-terra/qiskit/transpiler/passmanager.py:278, in PassManager._run_single_circuit(self, circuit, output_name, callback)
    266 """Run all the passes on a ``circuit``.
    267 
    268 Args:
   (...)
    275     The transformed circuit.
    276 """
    277 running_passmanager = self._create_running_passmanager()
--> 278 result = running_passmanager.run(circuit, output_name=output_name, callback=callback)
    279 self.property_set = running_passmanager.property_set
    280 return result

File ~/projects/qiskit-terra/qiskit/transpiler/runningpassmanager.py:124, in RunningPassManager.run(***failed resolving arguments***)
    122 for passset in self.working_list:
    123     for pass_ in passset:
--> 124         dag = self._do_pass(pass_, dag, passset.options)
    126 circuit = dag_to_circuit(dag)
    127 if output_name:

File ~/projects/qiskit-terra/qiskit/transpiler/runningpassmanager.py:155, in RunningPassManager._do_pass(self, pass_, dag, options)
    153 # Run the pass itself, if not already run
    154 if pass_ not in self.valid_passes:
--> 155     dag = self._run_this_pass(pass_, dag)
    157     # update the valid_passes property
    158     self._update_valid_passes(pass_)

File ~/projects/qiskit-terra/qiskit/transpiler/runningpassmanager.py:209, in RunningPassManager._run_this_pass(self, pass_, dag)
    206 elif pass_.is_analysis_pass:
    207     # Measure time if we have a callback or logging set
    208     start_time = time()
--> 209     pass_.run(FencedDAGCircuit(dag))
    210     end_time = time()
    211     run_time = end_time - start_time

File ~/projects/qiskit-terra/qiskit/transpiler/passes/layout/full_ancilla_allocation.py:68, in FullAncillaAllocation.run(self, dag)
     65 if layout is None:
     66     raise TranspilerError('FullAncillaAllocation pass requires property_set["layout"].')
---> 68 virtual_bits = layout.get_virtual_bits()
     69 physical_bits = layout.get_physical_bits()
     70 if layout:

AttributeError: 'list' object has no attribute 'get_virtual_bits'
```

### What should happen?

No error.

### Any suggestions?

_No response_

**Comment by kevinsung** [comment]
Ok it's because `initial_layout` is not allowed to be a list. I think it should be allowed to be any type accepted by `transpile` itself. I've changed the title of this issue accordingly.

**Comment by jakelishman** [comment]
Yeah, this is definitely part of a large interface problem surrounding the (complete lack of) separation of responsibilities between `PassManagerConfig` and `transpile`.  Really, `PassManagerConfig` isn't a nice interface as things stand, so the solution to this issue will likely come as larger changes to the whole pipeline come through (see e.g. #6403 for some work, where a similar issue is in the review comments).

A rework of this is definitely on the cards, though when it'll actually materialise is slightly difficult to say.
